MapboxNavigation/NavigationMapView.swift

```diff
@@ -252,6 +252,10 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
         
         installUserCourseView()
         showsUserLocation = false
+        
+        if let firstRoute = routes?.first {
```

Comment:
In case of `commonInit()` `routes` property will not be initialized yet, this means that this check will always fail and is not really needed. Let's test it to make sure and remove if it's not really needed.

Reply:
yes. So in case of it failed, the `updateRoute` did a check for the `fractionTraveled` calculated from the new method. If it failed, the `fractionTraveled` is default to 0.0, and the route line will be based on the old `routeProgress.fractionTraveled`. In the test, sometimes the `initPrimaryRoutePoints(route:)` has some problem during the initialization, so the check should be necessary. I'll look into this problem.